src/components/Common/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,207 @@
+    position: "relative",
+    padding: 0,
+  },
+  box: {
+    paddingBottom: 1,
+    paddingTop: 0,
+    paddingX: 2,
+  },
+  listBox: {
+    maxHeight: "60vh",
+  },
+  searchIcon: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "gray.main",
+    },
+  },
+  searchIconFocused: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "primary.main",
+    },
+  },
+};
+
+const SearchBarDropDown = ({ placeholder }: SearchBarDropDownProps) => {
+  const [focusSearchBar, setFocusSearchBar] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [keywords, setKeywords] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [searchOptions, setSearchOptions] = useState<SelectedUser[]>([]);
+  const { getAccessTokenSilently } = useAuth0();
+  const navigate = useNavigate();
+  const debouncedFetch = useMemo(
+    () => debounce((keywords: string) => fetchUsers(keywords), 100),
+    []
+  );
+
+  const fetchUsers = async (keywords: string) => {
+    setLoading(true);
+    if (keywords.length === 0) {
+      return;
+    }
+
+    try {
+      const token = await getAccessTokenSilently();
+      const result = await axios.get(
+        `http://localhost:3001/api/users/searchUsers`,
+        {
+          params: {
+            keyword: keywords,
+          },
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+
+      setSearchOptions(result.data as SelectedUser[]);
+
+      if (keywords.trim() === "") {
+        setSearchOptions([]);
+      }
+    } catch (error) {
+      console.error("Failed to fetch users:", error);
+      setSearchOptions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    debouncedFetch(keywords);
+  }, [keywords]);
+
+  const handleInputChange = (newInputValue: string) => {
+    setKeywords(newInputValue);
+
+    if (newInputValue.trim() === "") {
+      setSearchOptions([]);
+    }
+  };
+
+  const handleSelect = (selectedUsername: string) => {
+    setKeywords("");
+    setSearchOptions([]);
+    inputRef.current?.blur();
+    navigate(`/${selectedUsername}`);
+  };
+
+  const handleClear = () => {
+    setKeywords("");
+    setSearchOptions([]);
+    setFocusSearchBar(false);
+  };
+
+  return (
+    <Box sx={styles.box}>
+      <Autocomplete
+        disablePortal
+        fullWidth
+        filterOptions={(x) => x}
+        getOptionLabel={() => ""}
+        forcePopupIcon={false}
+        id="search"
+        value={null}
+        inputValue={keywords}
+        ListboxProps={{ style: styles.listBox }}
+        loading={!keywords && loading}
+        loadingText="Start typing to search..."
+        onChange={(_, value) => value && handleSelect(value.username)}
+        onBlur={handleClear}
+        onInputChange={(_, newInputValue) => handleInputChange(newInputValue)}
+        onFocus={() => setFocusSearchBar(true)}
+        open={focusSearchBar}
+        openOnFocus
```

Comment:
I think we only need one of these. If we're using `openOnFocus`, we shouldn't need to set the `open` prop, and vice versa